src/test/javascript/spec/component/import/exercise-import-from-file.component.spec.ts

```diff
@@ -0,0 +1,252 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { ExerciseImportFromFileComponent } from 'app/exercises/shared/import/from-file/exercise-import-from-file.component';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { ArtemisTestModule } from '../../test.module';
+import { HelpIconComponent } from 'app/shared/components/help-icon.component';
+import { MockComponent, MockDirective } from 'ng-mocks';
+import { ButtonComponent } from 'app/shared/components/button.component';
+import { ProgrammingExercise, ProgrammingLanguage, ProjectType } from 'app/entities/programming-exercise.model';
+import { Exercise, ExerciseType } from 'app/entities/exercise.model';
+import { AlertService } from 'app/core/util/alert.service';
+import { TranslateDirective } from 'app/shared/language/translate.directive';
+import JSZip from 'jszip';
+
+describe('ExerciseImportFromFileComponent', () => {
+    let component: ExerciseImportFromFileComponent;
+    let fixture: ComponentFixture<ExerciseImportFromFileComponent>;
+    let activeModal: NgbActiveModal;
+    let alertService: AlertService;
+    let alertServiceSpy: jest.SpyInstance;
+
+    beforeEach(async () => {
+        await TestBed.configureTestingModule({
+            imports: [ArtemisTestModule, MockComponent(HelpIconComponent), MockComponent(ButtonComponent)],
+            declarations: [ExerciseImportFromFileComponent, MockDirective(TranslateDirective)],
+        }).compileComponents();
+
+        fixture = TestBed.createComponent(ExerciseImportFromFileComponent);
+        component = fixture.componentInstance;
+        activeModal = TestBed.inject(NgbActiveModal);
+        alertService = TestBed.inject(AlertService);
+
+        fixture.detectChanges();
+    });
+
+    it('should create', () => {
+        expect(component).toBeTruthy();
+    });
+    it('should close the active modal with result', () => {
+        // GIVEN
+        const activeModalSpy = jest.spyOn(activeModal, 'close');
+        const exercise = { id: undefined } as ProgrammingExercise;
+        // WHEN
+        component.openImport(exercise);
+
+        // THEN
+        expect(activeModalSpy).toHaveBeenCalledOnce();
+        expect(activeModalSpy).toHaveBeenCalledWith(exercise);
+    });
+
+    // using fakeasync and tick didn't work here, that's why I used whenStable and async
+    it('should raise error alert if not supported exercise type', async () => {
```

Comment:
I remember that there was some issue quite a while ago due to which we switched to using `fakeAsync(() => {})` and `tick();` instead of `async` for all tests where possible.

(`async` tests not being properly awaited and the end and not being executed properly at all, therefore never failing?)